src/scandir.h

```diff
@@ -44,7 +44,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define dirent direct
 #endif
 
-#if defined(__BEOS__) || defined(__HAIKU__) || (defined(__sun) && defined(__SVR4)) || defined(WIN32)
```

Comment:
Seems alright but explain like I'm 5? Is `scandir` just a builtin on Haiku?

Reply:
I can't explain exactly how it has been, in the past there was a problem using scandir, this patch has been laying around for quite some time (not a developer here, but I've been seeing this in the past and not needed now).